src/main/scala/PacketAssembler/disassembler.scala

```diff
@@ -250,7 +250,7 @@ class PacketDisAssembler extends Module {
       counter := counterOut
       counter_byte := counterByteOut
     }.elsewhen(state === wait_dma) {
-      when (io.out.data.ready === true.B) {
+      when (io.out.length.ready === true.B && io.out.flag_aa.ready === true.B && io.out.flag_crc.ready === true.B) {
```

Comment:
Not sure what exactly this does, maybe check with @Fibonacci12358

Reply:
that is to be consistent with the RFtoDMAShim module

Reply:
Using all three ready flags like this means that this doesn't behave as 3 ready valid interfaces. This assumes that all 3 transactions happen during the same clock cycle, which is not necessarily true.
You need to add some extra logic to keep track of which outputs have been fired and go to idle when all have fired

Reply:
yeah you're right. I'm not sure my logic is completely correct but it should be fine with the current RFDMAshim?